### PR TITLE
LT-5623: [OrderBook] Add api to fetch last non zero spread by assetId + Save it in broker

### PR DESCRIPTION
**Pull request by lykke-vashetsin**

### CODE REVIEW

File: src/MarginTrading.OrderBookService.OrderBookBroker/Application.cs [hunk 1]
```diff
@@ -83,6 +88,20 @@ private Task HandleMessageWithoutThrottling(ExternalExchangeOrderbookMessage ord
...
+                {
+                    if (orderBook.Asks[0].Price != 0 && orderBook.Bids[0].Price != 0 &&
+                        orderBook.Asks[0].Price != orderBook.Bids[0].Price &&
+                        !string.IsNullOrWhiteSpace(orderBook.AssetPairId))
+                    {
+                        var spread = orderBook.Asks[0].Price - orderBook.Bids[0].Price;
+                        await _lastNonZeroSpreadService.Update(orderBook.AssetPairId, spread);
+                    }
+                }
+                catch (Exception ex)
```
Comment (tarurar) [inline src/MarginTrading.OrderBookService.OrderBookBroker/Application.cs:101]: What kind of exception do you expect here? You can't expect just "any" exception.

File: src/MarginTrading.OrderBookService.Services/LastNonZeroSpreadService.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+using StackExchange.Redis;
+
+namespace MarginTrading.OrderBookService.Services
+{
+    public class LastNonZeroSpreadService: ILastNonZeroSpreadService
+    {
+        private const string RedisLastNonZeroSpreadKeyFmt = "orderbook:asset:{0}:last-non-zero-spread";
+        private readonly IConnectionMultiplexer _redis;
+        private ILastNonZeroSpreadService _lastNonZeroSpreadServiceImplementation;
```
Comment (tarurar) [inline src/MarginTrading.OrderBookService.Services/LastNonZeroSpreadService.cs:19]: Did you forget to remove the line? Or was it an attempt to decorate?

File: src/MarginTrading.OrderBookService.OrderBookBroker/Startup.cs [hunk 3]
```diff
@@ -32,7 +34,10 @@ protected override void RegisterCustomServices(ContainerBuilder builder, IReload
             builder.AddMessagePackBrokerMessagingFactory<ExternalExchangeOrderbookMessage>();
             builder.RegisterType<Application>().As<IBrokerApplication>().SingleInstance();
             builder.RegisterType<SystemClock>().As<ISystemClock>().SingleInstance();
+            builder.RegisterType<LastNonZeroSpreadService>()
```
Comment (tarurar) [inline src/MarginTrading.OrderBookService.OrderBookBroker/Startup.cs:37]: Do you need it in both components: Broker and API?

File: src/MarginTrading.OrderBookService.Services/LastNonZeroSpreadService.cs (hunk 2 above)
Comment (lykke-vashetsin) [reply src/MarginTrading.OrderBookService.Services/LastNonZeroSpreadService.cs:19]: removed

File: src/MarginTrading.OrderBookService.OrderBookBroker/Startup.cs (hunk 3 above)
Comment (lykke-vashetsin) [reply src/MarginTrading.OrderBookService.OrderBookBroker/Startup.cs:37]: Api uses it to get data, broker - to update

File: src/MarginTrading.OrderBookService.OrderBookBroker/Application.cs (hunk 1 above)
Comment (lykke-vashetsin) [reply src/MarginTrading.OrderBookService.OrderBookBroker/Application.cs]: removed hanlder

**Review by tarurar — APPROVED** [review]

**Pull request merged**